### PR TITLE
luci-app-wifischedule: Convert to JS

### PR DESCRIPTION
**Pull request by Ramon00**
luci-app-wifischedule: Convert to JS

Signed-off-by: Ramon Van Gorkom Ramon00c00@gmail.com

### CODE REVIEW

**Comment by Ramon00** [comment]
So I think its 99% working, although I could not fully test it, seems the app on master has been updated compared to 23.05 branch and I have only devices on 23.05 currently running. The one thing that for sure does not work is the "Determine Modules Automatically" button. I am not sure how to get that working. It may be related to "ListValue.write(self, section, value_list)". I am not sure to convert that to js.  The issue is that the button from a option wants to update the content of another option. Look at line 148.  Does anybody have a suggestion?

**Comment by Ramon00** [comment]
@newkit Maybe you can help/test the js version?

**Comment by Ramon00** [comment]
Ok I changed the part that did not work, into a "display". Need to manually copy/paste it.  I will put this "for review" now. If somebody has a box running on master, please see if it fully works as I cannot fully test it as I only have things running 23.05.

File: applications/luci-app-wifischedule/htdocs/luci-static/resources/view/wifischedule/wifischedule.js [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+
+
+
+		o = s.option(form.Flag, 'enabled', _('Enable Wifi Schedule'));
+		o.optional = false;
+		o.rmempty = false;
+		o.validate = function(section_id, value) {
+			if (value == '1') {
+				if ((checkFile('/sbin/wifi')) && 
+					(checkFile('/usr/bin/wifi_schedule.sh'))) {
```
Comment (systemcrash) [inline applications/luci-app-wifischedule/htdocs/luci-static/resources/view/wifischedule/wifischedule.js:89]: This is OK but I'm thinking we can perform this check in a Promise at load time. SOP for many modules.

File: applications/luci-app-wifischedule/htdocs/luci-static/resources/view/wifischedule/wifischedule.js [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+		o.value('14:00');
+		o.value('15:00');
+		o.value('16:00');
+		o.value('17:00');
+		o.value('18:00');
+		o.value('19:00');
+		o.value('20:00');
+		o.value('21:00');
+		o.value('22:00');
+		o.value('23:00');
```
[the remark this answers is not in the document]
Comment (Ramon00) [reply applications/luci-app-wifischedule/htdocs/luci-static/resources/view/wifischedule/wifischedule.js:185]: let me try that. I just copied it from the original like it is now

File: applications/luci-app-wifischedule/htdocs/luci-static/resources/view/wifischedule/wifischedule.js (hunk 1 above)
Comment (Ramon00) [reply applications/luci-app-wifischedule/htdocs/luci-static/resources/view/wifischedule/wifischedule.js:89]: Let me check how other aps do it!

**Comment by systemcrash** [comment]
BTW just rebase (remove that merge commit please)

**Comment by Ramon00** [comment]
yeah will do

File: applications/luci-app-wifischedule/htdocs/luci-static/resources/view/wifischedule/wifischedule.js [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+
+		s = m.section(form.TypedSection, 'global', _('Manual control'));
+		s.optional = false;
+		s.rmempty = false;
+		s.anonymous = true;
+
+		o = s.option(form.Button, '', _('Activate wifi'));
+		o.onclick = function (section, value) {
+			fs.exec('/usr/bin/wifi_schedule.sh', ['start', 'manual']);
+			fs.exec('echo', ['a', '>/a.txt']);
```
Comment (systemcrash) [inline applications/luci-app-wifischedule/htdocs/luci-static/resources/view/wifischedule/wifischedule.js:61]: should this be here?

Comment (Ramon00) [reply applications/luci-app-wifischedule/htdocs/luci-static/resources/view/wifischedule/wifischedule.js]: nope, was for testing...removed now

File: applications/luci-app-wifischedule/htdocs/luci-static/resources/view/wifischedule/wifischedule.js [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+
+		o = s.option(form.TextValue, 'modules', 'Modules to unload')
+		o.depends('unload_modules', '1');
+		o.wrap = 'off';
+		o.rows = 10;
+		o.cfgvalue = function (section) {
+			let mod = uci.get('wifi_schedule', section, 'modules');
+			if (mod === undefined || mod === null) {
+			  mod = "";
+			}
```
Comment (systemcrash) [inline applications/luci-app-wifischedule/htdocs/luci-static/resources/view/wifischedule/wifischedule.js:111]: maybe just 
```js
if(!mod) mod = "";
```

File: applications/luci-app-wifischedule/htdocs/luci-static/resources/view/wifischedule/wifischedule.js [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+			return mod.replace(/ /g, "\r\n");
+		}
+		o.write = function (section, value) {
+				var valueList = value.replace(/[\r\n]+/g, " ").replace(/\s+$/, '');
+				return uci.set('wifi_schedule', section, 'modules', valueList);
```
Comment (systemcrash) [inline applications/luci-app-wifischedule/htdocs/luci-static/resources/view/wifischedule/wifischedule.js:116]: indent

File: applications/luci-app-wifischedule/htdocs/luci-static/resources/view/wifischedule/wifischedule.js [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+		}
+
+		o = s.option(form.TextValue, 'detectedmodules', _('Modules detected'));
```
Comment (systemcrash) [inline applications/luci-app-wifischedule/htdocs/luci-static/resources/view/wifischedule/wifischedule.js:119]: would this not be better as a `DummyValue`?

Comment (Ramon00) [reply applications/luci-app-wifischedule/htdocs/luci-static/resources/view/wifischedule/wifischedule.js]: Can do.

File: applications/luci-app-wifischedule/htdocs/luci-static/resources/view/wifischedule/wifischedule.js [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+			if (value == '0') {
+				if (data[3]) {
+					return true;
+				} else {
+					return _("Could not find required program /usr/bin/iwinfo");
+				}
+			} else {
+				return true;
+			}
+		};
```
Comment (systemcrash) [inline applications/luci-app-wifischedule/htdocs/luci-static/resources/view/wifischedule/wifischedule.js:178]:
```js
o.validate = function(section_id, value) {
    return value !== '0' || data[3] ? true : _("Could not find required program /usr/bin/iwinfo");
};
```

File: applications/luci-app-wifischedule/htdocs/luci-static/resources/view/wifischedule/wifischedule.js [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+			if (value == '1') {
+				if (data[0] && data[1]) {
+					return true;
+				} else {
+					return _('Could not find required /usr/bin/wifi_schedule.sh or /sbin/wifi');
+				}
+			} else {
+				return true;
+			}
+		};
```
Comment (systemcrash) [inline applications/luci-app-wifischedule/htdocs/luci-static/resources/view/wifischedule/wifischedule.js:91]:
```js
o.validate = function(section_id, value) {
    return value !== '1' || (data[0] && data[1]) ? true : _('Could not find required /usr/bin/wifi_schedule.sh or /sbin/wifi');
};
```

File: applications/luci-app-wifischedule/htdocs/luci-static/resources/view/wifischedule/wifischedule.js (hunk 7 above)
Comment (Ramon00) [reply applications/luci-app-wifischedule/htdocs/luci-static/resources/view/wifischedule/wifischedule.js:178]: I can change that.

File: applications/luci-app-wifischedule/htdocs/luci-static/resources/view/wifischedule/wifischedule.js (hunk 8 above)
Comment (Ramon00) [reply applications/luci-app-wifischedule/htdocs/luci-static/resources/view/wifischedule/wifischedule.js:91]: I can change that.

**Comment by Ramon00** [comment]
@systemcrash Any more comments? guess next time it is probably quicker and easier if you do the conversion yourself!

**Comment by systemcrash** [comment]
I think this is done. Looks great!

**Comment by systemcrash** [comment]
Thanks @Ramon00

**Comment by Ramon00** [comment]
@systemcrash More like thank you, fixing all the mistakes...

**Comment by hnyman** [comment]
If this has now been converted to Javascript, is there any need to luci-compat dependency any more?  It is supposed to be needed only for the lua support for old apps..

**Comment by systemcrash** [comment]
Nope. Wanna fix that?

**Pull request merged**